src/overlays/actors/ovl_Eff_Kamejima_Wave/z_eff_kamejima_wave.c

```diff
@@ -27,26 +34,138 @@ const ActorInit Eff_Kamejima_Wave_InitVars = {
+
+    Actor_SetScale(&this->actor, 0.2f);
+    this->actor.scale.y = 0.0f;
+    if (EFFKAMEJIMAWAVE_GET_F(thisx) == PARAMS_1) {
+        this->actionFunc = func_80BCEDE0;
+        D_80BCF1C0 = 255;
+    } else {
+        this->actionFunc = func_80BCEC6C;
+        D_80BCF1C0 = 255;
+    }
+    D_80BCF1C4 = Lib_SegmentedToVirtual(object_kamejima_Matanimheader_001AF0);
+}
+
+void EffKamejimaWave_Destroy(Actor* thisx, PlayState* play) {
+}
+
+void func_80BCEBC0(EffKamejimaWave* this, PlayState* play) {
+    this->actor.scale.y += this->scaleOffset;
+    this->scaleOffset -= 0.003f;
+    this->actor.shape.rot.x += 0x64;
+    if (this->scaleOffset < 0.0f) {
+        this->actor.scale.x += 0.003f;
+    }
+    if (this->scaleOffset < -0.01f) {
+        this->scaleOffset = -0.01f;
+    }
+    if (this->actor.scale.y < 0.0f) {
+        this->actor.scale.y = 0.0f;
+        this->actor.scale.x = 0.2f;
+        this->actionFunc = func_80BCEC6C;
+        this->actor.draw = NULL;
+    }
+}
+
+void func_80BCEC6C(EffKamejimaWave* this, PlayState* play) {
+    if (Cutscene_CheckActorAction(play, 0x1E0)) {
+        Cutscene_ActorTranslateAndYaw(&this->actor, play, Cutscene_GetActorActionIndex(play, 0x1E0));
+        if (play->csCtx.actorActions[Cutscene_GetActorActionIndex(play, 0x1E0)]->action == 2) {
+            this->scaleOffset = 0.05f;
+            this->actionFunc = func_80BCEBC0;
+            this->actor.draw = EffKamejimaWave_Draw;
+            D_80BCF1C0 = 255;
+            this->actor.shape.rot.x = 0;
+            Actor_PlaySfxAtPos(&this->actor, NA_SE_EV_TORTOISE_WAKE_UP);
+        }
+    } else {
+        this->actor.draw = NULL;
+    }
+}
+
+void func_80BCED34(EffKamejimaWave* this, PlayState* play) {
+    this->actor.scale.y += this->scaleOffset;
+    this->scaleOffset -= 0.002f;
+    this->actor.shape.rot.x += 0x8C;
+    if (this->scaleOffset < 0.0f) {
+        this->actor.scale.x += 0.002f;
+    }
+    if (this->scaleOffset < 0.0f) {
+        this->scaleOffset = 0;
+    }
+    if (D_80BCF1C0 >= 4) {
+        D_80BCF1C0 -= 4;
+        return;
+    }
+    this->actor.scale.y = 0.0f;
+    this->actionFunc = func_80BCEDE0;
+    this->actor.draw = NULL;
+}
+
+void func_80BCEDE0(EffKamejimaWave* this, PlayState* play) {
+    if (Cutscene_CheckActorAction(play, 0x1F6)) {
+        Cutscene_ActorTranslateAndYaw(&this->actor, play, Cutscene_GetActorActionIndex(play, 0x1F6));
+        if (play->csCtx.actorActions[Cutscene_GetActorActionIndex(play, 0x1F6)]->action == 2) {
+            this->scaleOffset = 0.03f;
+            this->actor.scale.x = 0.35f;
+            this->actionFunc = func_80BCED34;
+            this->actor.draw = EffKamejimaWave_Draw;
+            this->actor.shape.rot.x = 0;
+        }
+    } else {
+        this->actor.draw = NULL;
+    }
+}
+
+void EffKamejimaWave_Update(Actor* thisx, PlayState* play) {
+    EffKamejimaWave* this = THIS;
+
+    this->actionFunc(this, play);
+}
+
+void func_80BCEEBC(u8 alpha) {
```

Comment:
So `...AdjustVtxAlpha`?

Reply:
Personally I like Set over Adjust

Reply:
This comment was marked as "resolved" but it there are no changes. Unresolving and waiting for author's feedback

Reply:
Seems i renamed a wrong function from a wrong file, late night coding, apologize for that, thank you for pointing this out